openwisp_ipam/api/views.py

```diff
@@ -129,7 +129,12 @@ def __getitem__(self, i):
             return HostsSet(self.subnet, self.start + start, self.start + stop)
         if i >= self.count():
             raise IndexError
-        host = self.subnet.subnet._address_class(self.network + 1 + i + self.start)
+        # In case of single hosts ie subnet/32 & /128
+        if self.subnet.subnet.prefixlen in [32, 128]:
+            host = self.subnet.subnet._address_class(self.network + i + self.start)
+        else:
+            # Host starts from next address
+            host = self.subnet.subnet._address_class(self.network + 1 + i + self.start)
```

Comment:
great but we need a test for this new case please.

There's also some repetition here which can be avoided, avoid repeating the whole line `host = self.subnet.subnet._address_class(...)`

Reply:
> great but we need a test for this new case please.
> 
> There's also some repetition here which can be avoided, avoid repeating the whole line `host = self.subnet.subnet._address_class(...)`

Done :+1: